### PR TITLE
Don't use reentrant dispatch for composite compliance

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #84893
* __->__ #84909

I believe these were added in to prevent changing behavior when
https://github.com/pytorch/pytorch/pull/75827 landed, but I actually
think they are unnecessary, and they are causing asserts to fire
on the subsequent PR (where I assert that tensors returned by
views MUST NOT already have view metadata associated with them.)

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

**Review by zou3519 — APPROVED** [review]
> I believe these were added in to prevent changing behavior when
https://github.com/pytorch/pytorch/pull/75827 landed, but I actually
think they are unnecessary

Yup, agreed

File: torch/testing/_internal/composite_compliance.py [hunk 1]
```diff
@@ -197,38 +196,36 @@ def wrap(e):
                         'regular Tensor but the other tensors are Tensor Subclasses. '
                         'Please try to avoid this in-place operation.')
 
-            with enable_reentrant_dispatch():
-                with contextlib.nullcontext() if enable_recursive_torch_dispatch else no_dispatch():
-                    unwrapped_args = tree_map(unwrap, args)
-                    unwrapped_kwargs = tree_map(unwrap, kwargs)
-                    unwrapped_rs = func(*unwrapped_args, **unwrapped_kwargs)
-                    rs = tree_map(wrap, unwrapped_rs)
+            with contextlib.nullcontext() if enable_recursive_torch_dispatch else no_dispatch():
```
Comment (soulitzer) [inline torch/testing/_internal/composite_compliance.py:199]: Could it be possible to remove the enable_recursive_torch_dispatch too? It probably doesn't matter to leave it on either for non-mode python dispatch.

**Review by soulitzer — APPROVED** [review]
Nice. I think this may resolve some weirdness with forward AD CCT as well - previously it was probably doing forward AD twice.

File: torch/testing/_internal/composite_compliance.py (hunk 1 above)
Comment (ezyang) [reply torch/testing/_internal/composite_compliance.py:199]: I'll test that on a separate PR

**Pull request closed without merging**